packages/vela/src/Vela.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  type MintActionParams,
+  type TransactionFilter,
+  type SwapActionParams,
+  type StakeActionParams,
+  compressJson,
+} from '@rabbitholegg/questdk'
+import { type Address } from 'viem'
+import { VAULT_CONTRACT, CHAIN_TO_TOKENS, TOKENFARM_CONTRACT, VLP_CONTRACT, VELA_CONTRACT } from './contract-addresses'
+import { CHAIN_ID_ARRAY } from './chain-ids'
+import { TOKENFARM_ABI, TOKENFARM_ABI2, VAULT_ABI } from './abi'
+
+export const swap = async (
+  swap: SwapActionParams,
+): Promise<TransactionFilter> => {
+  const { chainId, contractAddress, tokenIn, amountIn, amountOut, deadline, recipient } = swap
+
+  return compressJson({
+    chainId,
+    to: contractAddress ?? VAULT_CONTRACT,
+    from: recipient,
+    input: {
+      $abi: VAULT_ABI,
+      a: amountIn,
```

Comment:
These values are packed, and will not match as you have them here.

I made a couple helper functions for unpacking the data that could help.

https://github.com/rabbitholegg/questdk-plugins/blob/f030d3e58474e46b0df73ba372009238797919b5/packages/vela/src/utils.ts#L68

Reply:
Sorry but we cant use your way. 
![image](https://github.com/rabbitholegg/questdk-plugins/assets/56060401/16c1ee94-4339-44c2-b8a1-d3101527398b)
You can check our newPositionOrderPacked function.

Reply:
@lovetrading10 

I just finished submitting a pr to take care of the derivatives action.

https://github.com/rabbitholegg/questdk-plugins/pull/129


We were able to unpack the values using a bitmask.